src/Illuminate/Support/Collection.php

```diff
@@ -91,6 +91,17 @@ public function collapse()
     }
 
     /**
+     * Concat the collection with the given items.
+     *
+     * @param  mixed  $items
+     * @return static
+     */
+    public function concat($items)
+    {
+        return new static($this->getArrayableItems($items) + $this->items);
```

Comment:
Also you are calling `\Illuminate\Support\Collection::getArrayableItems` twice:
- once on `$items` in `\Illuminate\Support\Collection::concat`
- the same subset will be pass through `getArrayableItems` again in `\Illuminate\Support\Collection::__construct`